pkg/controller/cyclenoderequest/transitioner/util.go

```diff
@@ -136,7 +141,8 @@ func (t *CycleNodeRequestTransitioner) reapChildren() (v1.CycleNodeRequestPhase,
 	}
 
 	// If we've finished most of our children, go back to Initialised to add some more nodes
-	if t.cycleNodeRequest.Status.ActiveChildren <= t.cycleNodeRequest.Spec.CycleSettings.Concurrency/2 {
+	// If the next phase should be failed, skip this since transitioning back to initialised would be flip-flopping behaviour
+	if nextPhase != v1.CycleNodeRequestFailed && t.cycleNodeRequest.Status.ActiveChildren <= t.cycleNodeRequest.Spec.CycleSettings.Concurrency/2 {
```

Comment:
How does `t.cycleNodeRequest.Status.ActiveChildren <= t.cycleNodeRequest.Spec.CycleSettings.Concurrency/2` work? PArticular why divide concurrency by 2?

Reply:
If at least half of the node in the current cycle have finished cycling, then cyclops heads back to `Initialised` to pick up more. Problem here is that if there's a single node selected in the `CNR` and the draining times out, the `CNS` object gets deleted and `ActiveChildren` stays at 0 which will always trigger going back to `Initialised`.

Reply:
@Jacobious52 `t.cycleNodeRequest.Status.ActiveChildren <= t.cycleNodeRequest.Spec.CycleSettings.Concurrency/2` allows us to pick up more work when the majority of nodes in the current batch of `concurrency` have been cycled.

If concurrency is 3, rather than waiting for the entire 3 to finish before starting more, we wait for the majority to finish, i.e. 2 before starting more cycling. 

It's a minor optimisation, but for larger concurrency numbers would make a difference in speed.